epsagon/trace.py

```diff
@@ -321,6 +397,11 @@ def pop_trace(self, trace=None):
         :return: unique id
         """
         with self.LOCK:
+            if self.use_async_tracer:
+                trace = self._pop_trace_async_mode()
+                if trace:
+                    # async tracer found
+                    return trace
```

Comment:
actually, in this case - use async tracer = True. If no trace has been found, there's no need to continue checking other states - can just return the ret value from pop_trace_async_mode